realm/src/main/java/io/realm/internal/ColumnIndicies.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2014 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import io.realm.RealmObject;
+
+/**
+ * Utility class used to cache the mapping between object field names and their column indices.
+ */
+public class ColumnIndicies {
```

Comment:
Because the order is based on the order of the fields in the source code. In order to create provide compile time constants you would also require migrations if you changed the order of the fields, and you would have to ensure the same order across platforms. That would be a horrible user experience.

So you are right that the order is fixed once we have run validateTable, but it still has to be calculated at runtime which means we cannot use constants.